### PR TITLE
【new_frl】modify multiply_grad_node create

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types

Function optimization
### PR changes

Others
### Description

pcard-66975
修改multiply创建一阶反向时可根据输入是否为stop_gradient 确认是否hold 前向输入数据。

### CODE REVIEW

File: paddle/fluid/eager/api/manual/eager_manual/dygraph_forward_api.h [hunk 1]
```diff
@@ -26,6 +26,10 @@ paddle::Tensor conv2d_ad_func(const paddle::Tensor& input,
                               std::vector<int> dilations,
                               int groups,
                               std::string data_format);
+paddle::Tensor multiply_ad_func(const paddle::Tensor& x,
+                                const paddle::Tensor& y);
+paddle::Tensor& multiply__ad_func(paddle::Tensor& x,  // NOLINT
```
Comment (JiabinYang) [inline paddle/fluid/eager/api/manual/eager_manual/dygraph_forward_api.h:31]: why inplace?

**Review by JiabinYang — COMMENTED** [review]
One comments

File: paddle/fluid/eager/api/manual/eager_manual/dygraph_forward_api.h (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/eager/api/manual/eager_manual/dygraph_forward_api.h:31]: inplace can not be generated because of 'multiply' added in black list

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**